.eslintrc

```diff
@@ -1,15 +1,39 @@
 {
     "root": true,
     "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "project": "./tsconfig.json"
+    },
     "plugins": ["@typescript-eslint"],
     "extends": [
         "eslint:recommended",
         "plugin:@typescript-eslint/eslint-recommended",
-        "plugin:@typescript-eslint/recommended"
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking"
     ],
     "env": {
         "browser": true,
         "node": true
     },
-    "rules": {}
+    "rules": {
+        "func-style": "error",
+        "prefer-arrow-callback": "error",
+        "@typescript-eslint/unbound-method": "off", // TODO: Do we want / need this?
```

Comment:
what does it do? haha

Reply:
I don't really understand TBH (based off of what it complained about and reading through the docs). The docs make it seem like it is related to classes but I'd think it wouldn't complain about in our code in that case. 🤷 Maybe you can have better luck understanding what it is: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/unbound-method.md

Reply:
Looks like it has to do with how `this` is resolved depending on how you unpack/destructure a function from an object. Since we're not using classes, and we're not using `this` anywhere, I don't _think_ this rule really applies to this code base?

Reply:
https://github.com/taylorn16/fp-toolkit/pull/31/commits/c94537d7d64497bb5868efb2848072215f83deb0